src/module.rs

```diff
@@ -40,6 +40,31 @@ pub enum Linkage {
     WeakODRLinkage,
 }
 
+/// Defines the optimization level used to compile a `Module`.
+/// 
+/// # Remarks
+/// See also: http://llvm.org/doxygen/CodeGen_8h_source.html
+#[derive(Debug, PartialEq, Eq, Copy, Clone)]
+pub enum OptimizationLevel {
```

Comment:
This enum already exists here under target.rs: https://github.com/TheDan64/inkwell/blob/cdc1560f4c247698684204d563b8a88b3fded2d3/src/targets.rs#L17-L21

Assigning values is definitely a good idea, but I think `Option<OptimizationLevel>` is a bit more rusty. It's easy enough to use internally:  `optimization_level_opt.unwrap_or(0) as i32` and I feel like it's easier for API users, since they can just toss in a `None` if they don't actually know what it does, and are just experimenting for example. Otherwise they have to import the enum and look up the variants.

Any thoughts on this? I'm alright with renaming `CodeGenOptLevel` to `OptimizationLevel` if you think that's a better name. I'm not quite sure where this should live though now that it's needed in two places

Reply:
Oh, I didn't see, sorry. As you wish; it looks like most members in Inkwell are more verbose than in LLVM (for example: `build_float_add` vs `CreateFAdd`), which is why I spelled `OptimizationLevel` completely.

Is the fact that it is more "rusty" good? If so, that's actually what I used.  
In any case, choose what you find is best. To me, mapping enum values is better since it's a feature, and it allows to put all the logic in the definition of the enum, rather than on in the functions that use it / in a trait.

Reply:
For names I've mostly been snake_casing llvm-sys names, lol. Keeping the logic in one place is a good point. Just trying to think about how the API will be used. Hmm.

Reply:
Oh, and since `OptimizationLevel` is an `u32`, it's faster to pass it by value (via copying) rather than by reference. I'm pushing the new changes now, and using my solution, at least for now.